microsetta_public_api/api/tests/test_api.py

```diff
@@ -0,0 +1,34 @@
+from unittest import TestCase
+from unittest import mock
+
+import microsetta_public_api
+import microsetta_public_api.server
+
+
+class FlaskTests(TestCase):
+
+    def setUp(self):
+
+        self.app, self.client = self.build_app_test_client()
+
+    @staticmethod
+    def build_app_test_client():
+        app = microsetta_public_api.server.build_app()
+        client = app.app.test_client()
+        return app, client
+
+
+class AlphaDiversityTests(FlaskTests):
+
+    def test_alpha_diversity_single_sample(self):
+
+        ret_val = {}
+
+        with mock.patch('microsetta_public_api.api.diversity.alpha.get_alpha'
+                        '') as mocked_alpha:
+            mocked_alpha.return_value = (ret_val, 72419)
+            _, client = self.build_app_test_client()
+            response = client.get(
+                '/api/diversity/alpha/observed_otus/sample-foo-bar')
+
+        self.assertEqual(response.status_code, 72419)
```

Comment:
the status code is 72419?

Reply:
So since we haven't actually determined where/how alpha is getting its information, I mocked a status code. The mocked status code should not be 404, for obvious reasons. I also figured mocking a status code that was non-standard, then checking its value could help verify that the specific function I think is getting alpha diversity, is actually getting it. Whereas if I mocked the status code as 200, it theoretically could be getting that code from any number of places. So 72419 was a unique value it was unlikely to get from any other functions.

Reply:
Sorry for slow reply. `200` would be correct, and I don't think we want to test whether flask + connexion are working, but rather that our endpoints are working. Does that make sense?

Reply:
Yeah I guess it makes more sense once we have actual data that we are operating on, or more of an interface built out. I have changed it to test for `200` status code and just whether it gets back the fake dictionary I told it to return for the time being.
See: https://github.com/biocore/microsetta-public-api/pull/1/commits/59d8d09fb2dcfa5a3d42622eca975ed7b3836b1e